R/utils.R

```diff
@@ -210,3 +208,44 @@ convert <- function(x) {
   x[y]      <- converted
   x
 }
+
+#' Add a report obtained from a data cleaning step to a data frame
+#'
+#' @param data the input data frame
+#' @param report a named `list` or `data frame` containing details from the
+#'    cleaning step
+#' @param name a `character` with the name of the cleaning step. The default
+#'    value is NULL.
+#'
+#' @return the input data frame with a report associated to it. This can be
+#'    accessed using `attr(data, "report")`
+#' @keywords internal
+#' @noRd
+#'
+add_report <- function(data, report, name = NULL) {
```

Comment:
```suggestion
add_report <- function(data, report, name = NULL) {
```

Consider removing the `name` parameter from the function signature and adding a `name` field to `report` in the cleaning functions. If that is done this function can be implemented as a concatenation of reports from each cleaning step to the list/dataframe in the `report` attribute of `data`.

Reply:
The `add_report()` function is an internal function that is called within the different functions for cleaning.
The value for the `name` argument is hardcoded inside the function calling `add_report()`.

The advantage of having the `name` argument is that it allows to either create a new element into the report object of concatenate the new cleaning report to an existing one. The `name` argument helps in telling exactly about which element of the report to create or update.

Reply:
Thanks for the clarification.